### PR TITLE
feat: convert Message to functional style

### PR DESCRIPTION
**Pull request by tjwelde**
## fixes KILTProtocol/ticket#2164
This PR changes the Message class to the functional programming style.

## TODO
- [x] Move message.utils functions into base message file

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/messaging/src/Message.ts [hunk 1]
```diff
@@ -26,312 +25,287 @@ import { DidResolver } from '@kiltprotocol/did'
...
+ * Constructs a message from a message body.
+ * This should be encrypted with [[Message.encrypt]] before sending to the receiver.
+ *
+ * @param body The body of the message.
+ * @param sender The DID of the sender.
+ * @param receiver The DID of the receiver.
+ * @returns The message created.
+ */
+export function fromBody(
+  body: MessageBody | CompressedMessageBody,
```
Comment (tjwelde) [inline packages/messaging/src/Message.ts:213]: The compression for messages seems to be not thought through. You can only compress the message body (not a full message) and if you try to make a new `Message` object from a compressed message body, it is decompressed, before making the message.
Seems like a compressed message body can never really be used for message sending.

Comment (tjwelde) [reply packages/messaging/src/Message.ts:213]: I would like to at least get rid of the option to give a compressed message body to this function. What d you think?

Comment (arty-name) [reply packages/messaging/src/Message.ts:213]: If it is so half-baked, let’s rather drop it :+1:

Comment (tjwelde) [reply packages/messaging/src/Message.ts]: I also added a ticket for getting rid of compression altogether

**Review by arty-name — APPROVED** [review]

File: packages/messaging/src/Message.ts [hunk 2]
```diff
@@ -5,330 +5,525 @@
...
-        }
-        break
-      default:
-    }
+/**
+ * Checks if delegation data is well formed.
+ *
+ * @param delegationData Delegation data to check.
+ */
+export function errorCheckDelegationData(
```
[the remark this answers is not in the document]
Comment (tjwelde) [reply packages/messaging/src/Message.ts:47]: Yeah, there is even more i would like to move / cleanup. Will do in another ticket though

**Pull request merged**